xlive/Blam/Engine/game/cheats.cpp

```diff
@@ -0,0 +1,98 @@
+#include "stdafx.h"
+#include "cheats.h"
+
+#include "Blam/Engine/tag_files/global_string_ids.h"
+#include "Blam/Engine/game/game_globals.h"
+
+#include "H2MOD/Tags/TagInterface.h"
+
+#include "Util/Hooks/Hook.h"
+
+
+typedef bool(__cdecl* ice_cream_flavor_available_t)(unsigned int skull);
+ice_cream_flavor_available_t p_ice_cream_flavor_available;
+
+typedef void(__cdecl* ice_cream_activation_t)(unsigned int skull);
+ice_cream_activation_t p_ice_cream_activation;
+
+bool skull_enabled[SKULL_COUNT] = { false };
+const string_id skull_string_ids[SKULL_COUNT] =
+{
+	HS_UNLOCKED_EXTRA_1,	// Envy
+	HS_UNLOCKED_EXTRA_2,	// Grunt Birthday Party
+	HS_UNLOCKED_EXTRA_3,	// Assassins
+	HS_UNLOCKED_EXTRA_4,	// Thunderstorm
+	HS_UNLOCKED_EXTRA_5,	// Famine
+	HS_UNLOCKED_EXTRA_6,	// I Would Have Been Your Daddy
+	HS_UNLOCKED_EXTRA_7,	// Blind
+	HS_UNLOCKED_EXTRA_8,	// Ghost
+	HS_UNLOCKED_EXTRA_9,	// Black Eye
+	HS_UNLOCKED_EXTRA_10,	// Catch
+	HS_UNLOCKED_EXTRA_11,	// Sputnik
+	HS_UNLOCKED_EXTRA_12,	// Iron
+	HS_UNLOCKED_EXTRA_13,	// Mythic
+	HS_UNLOCKED_EXTRA_14,	// Angry
+	HS_UNLOCKED_EXTRA_16,	// That's Just... Wrong
+	HS_UNLOCKED_EXTRA_15,	// Cow Bell
+};
+
+
+
+bool __cdecl ice_cream_flavor_available(const e_skull_type skull)
+{
+	return skull < SKULL_COUNT && s_game_globals::game_is_campaign() && skull_enabled[skull];
+}
+
+void __cdecl ice_cream_flavor_stock(const e_skull_type skull)
+{
+	typedef void(__cdecl* hud_clear_messages_t)();
+	hud_clear_messages_t p_hud_clear_messages = Memory::GetAddress<hud_clear_messages_t>(0x22CE83, 0x206863);
```

Comment:
`auto` because it's pointless to do it twice

Reply:
changed